cdap-common/src/main/java/io/cdap/cdap/common/metadata/QueryParser.java

```diff
@@ -109,23 +113,78 @@ public static String extractTermValue(String term) {
   }
 
   private static QueryTerm parseQueryTerm(String term) {
-    SearchType parsedSearchType = getSearchType(term);
+    Qualifier parsedQualifier = hasRequirementPrefix(term) ? Qualifier.REQUIRED : Qualifier.OPTIONAL;
+    String parsedTerm = hasRequirementPrefix(term)
+      ? term.substring(1) : term;
```

Comment:
nit: move this to line 117.

Reply:
Do you mean move `SearchType parsedSearchType = getSearchType(term);` to right before line 117? 

Right now I call that line after the if statement in line 120 because parsedSearchType is not needed for the return statement within the if-statement, and so would be computed unnecessarily.

But if that's not a big concern and it helps in readability, I can move it

Reply:
I mean `String parsedTerm = hasRequirementPrefix(term) ? ? term.substring(1) : term;`

Reply:
Ah I see, ok I can switch that

Reply:
missed?